### PR TITLE
Port missing Integration tests from EI

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
$subject

### CODE REVIEW

File: integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/ProxyStatisticsTest.java [hunk 1]
```diff
@@ -71,6 +71,39 @@ public void statisticsCollectionCountTest() throws Exception {
...
+
+        ArrayList<String> mediatorList = new ArrayList<>();
+        mediatorList.add("TestEndpointStatProxy@0:TestEndpointStatProxy");
+        mediatorList.add("TestEndpointStatProxy@1:PROXY_INSEQ");
+        mediatorList.add("TestEndpointStatProxy@2:CallMediator");
+        mediatorList.add("TestEndpointStatProxy@3:StatEndpoint1");
+        mediatorList.add("TestEndpointStatProxy@4:RespondMediator");
+
+        //Checking whether all the mediators are present in the event
+        Assert.assertEquals(eventList.size(), 5, "Four configuration events are required");
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/ProxyStatisticsTest.java:100]: four or five ?

File: integration/mediation-tests/tests-patches/src/test/resources/artifacts/ESB/passthru/transport/preserveResponseHeaders/passthru-http.properties [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Pass-through HTTP transport specific tuning parameters
+http.socket.timeout=180000
+
+worker_pool_size_core=400
+worker_pool_size_max=500
+#worker_thread_keepalive_sec=60
+#worker_pool_queue_length=-1
+#io_threads_per_reactor=2
+io_buffer_size=16384
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-patches/src/test/resources/artifacts/ESB/passthru/transport/preserveResponseHeaders/passthru-http.properties:27]: do we need this file, isn't this configured via toml ?

Comment (SanojPunchihewa) [reply integration/mediation-tests/tests-patches/src/test/resources/artifacts/ESB/passthru/transport/preserveResponseHeaders/passthru-http.properties:27]: The `passthru-http.properties.j2` file was not updated with this `http.response.headers.preserve` property. I have created an issue to track it https://github.com/wso2/api-manager/issues/34

File: integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/ProxyStatisticsTest.java (hunk 1 above)
Comment (SanojPunchihewa) [reply integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/ProxyStatisticsTest.java]: Updated the message with [5191446](https://github.com/wso2/micro-integrator/pull/2700/commits/519144676b3ab210de454c6e8570859c21cb535e)

**Review by Nirothipan — APPROVED** [review]

**Pull request merged**